pkg/security/tests/action_test.go

```diff
@@ -563,3 +563,74 @@ func TestActionKillDisarm(t *testing.T) {
 		})
 	})
 }
+
+func TestActionHash(t *testing.T) {
+	SkipIfNotAvailable(t)
+
+	ruleDefs := []*rules.RuleDefinition{
+		{
+			ID:         "hash_action",
+			Expression: `open.file.path == "{{.Root}}/test-hash-action" && open.flags&O_CREAT == O_CREAT`,
+			Actions: []*rules.ActionDefinition{
+				{
+					Hash: &rules.HashDefinition{},
+				},
+			},
+		},
+	}
+
+	test, err := newTestModule(t, nil, ruleDefs)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer test.Close()
+
+	testFile, _, err := test.Path("test-hash-action")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	syscallTester, err := loadSyscallTester(t, test, "syscall_tester")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("open", func(t *testing.T) {
+		test.WaitSignal(t, func() error {
+			go func() {
+				timeoutCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+				defer cancel()
+
+				if err := runSyscallTesterFunc(
+					timeoutCtx, t, syscallTester,
+					"slow-write", "2", testFile, "aaa",
```

Comment:
If I understand correctly, here you test the case were the process writing the file exits BEFORE the 5s defaultHashActionFlushDelay ?
I wonder if we should also test the case where it takes more than this delay to exit?